bundles/org.openhab.binding.meross/src/main/java/org/openhab/binding/meross/internal/api/MerossHttpConnector.java

```diff
@@ -0,0 +1,280 @@
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * The {@link MerossHttpConnector} class is responsible for handling the Http functionality for connecting to the Meross
+ * Cloud
+ *
+ * @author Giovanni Fabiani - Initial contribution
+ */
+@NonNullByDefault
+public class MerossHttpConnector {
+    private final Logger logger = LoggerFactory.getLogger(MerossHttpConnector.class);
+    private static final String INITIAL_STRING = "23x17ahWarFH6w29";
+    private static final String DEFAULT_APP_TYPE = "MerossIOT";
+    private static final String MODULE_VERSION = "0.0.0";
+    private static final long CONNECTION_TIMEOUT_SECONDS = 15;
+    private @Nullable String token;
+    private final String apiBaseUrl;
+    private final String userEmail;
+    private final String userPassword;
+    private final HttpClient client = HttpClient.newBuilder()
+            .connectTimeout(Duration.of(CONNECTION_TIMEOUT_SECONDS, ChronoUnit.SECONDS)).build();
+    private final File credentialFile;
+    private final File deviceFile;
+
+    public MerossHttpConnector(String apiBaseUrl, String userEmail, String password, File credentialFile,
+            File deviceFile) {
+        this.apiBaseUrl = apiBaseUrl;
+        this.userEmail = userEmail;
+        this.userPassword = password;
+        this.credentialFile = credentialFile;
+        this.deviceFile = deviceFile;
+    }
+
+    /**
+     * @param paramsData The params
+     * @param uri The uri
+     * @param path The path (endpoint)
+     * @return The http response
+     * @throws IOException if it fails to return the http response
+     */
+    private HttpResponse<String> postResponse(Map<String, String> paramsData, String uri, String path)
+            throws IOException {
+        String dataToSign;
+        String encodedParams;
+        String authorizationValue;
+        String nonce = UUID.randomUUID().toString().replace("-", "").substring(0, 16).toUpperCase();
+        long timestamp = Instant.now().toEpochMilli();
+        encodedParams = encodeParams(paramsData);
+        dataToSign = "%s%d%s%s".formatted(INITIAL_STRING, timestamp, nonce, encodedParams);
+        String md5hash = MD5Util.getMD5String(dataToSign);
+        Map<String, String> payloadMap = new HashMap<>();
+        payloadMap.put("params", encodedParams);
+        payloadMap.put("sign", md5hash);
+        payloadMap.put("timestamp", String.valueOf(timestamp));
+        payloadMap.put("nonce", nonce);
+        String payload = new Gson().toJson(payloadMap);
+        if (token != null) {
+            authorizationValue = "Basic %s".formatted(token);
+        } else {
+            authorizationValue = "Basic";
+        }
+        HttpRequest postRequest = HttpRequest.newBuilder().uri(URI.create(uri + path))
+                .header("Authorization", authorizationValue).header("AppVersion", "0.0.0").header("vender", "meross")
```

Comment:
vender or vendor? is it needed?

Reply:
I see but that is right probably a spelling issue as the server side!